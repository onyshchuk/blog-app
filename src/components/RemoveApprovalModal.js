import React from 'react';
import Modal from 'react-modal';

const RemoveApprovalModal = props => {
   const onRemoveClick = () => {
      props.handleRemoveClick();
      props.closeModal();
   }
   const onNoClick = () => {
      props.closeModal();
   }
   return (
      <Modal
         className="Modal"
         overlayClassName="Overlay"
         closeTimeoutMS={200}
         isOpen={props.isOpen}
         contentLabel={`Remove '${props.selectedPostTitle}' ?`}
         appElement={document.getElementById('app')}
         onRequestClose={props.closeModal}
      >
         <h3>Remove {props.selectedPostTitle} ?</h3>
         <div className="buttons">
            <button className="button" onClick={onRemoveClick}>Remove</button>
            <button className="button" onClick={onNoClick}>no</button>
         </div>
      </Modal>
   );
}

export default RemoveApprovalModal;