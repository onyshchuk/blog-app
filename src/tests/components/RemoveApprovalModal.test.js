import React from 'react';
import { shallow } from 'enzyme';
import RemoveApprovalModal from './../../components/RemoveApprovalModal';

let handleRemoveClick, closeModal, wrapper;

beforeEach(() => {
   closeModal = jest.fn();
   handleRemoveClick = jest.fn();
   wrapper = shallow(<RemoveApprovalModal 
      closeModal={closeModal}
      handleRemoveClick={handleRemoveClick}
   />);
});

test('should render RemoveApprovalModal properly', () => {
   expect(wrapper).toMatchSnapshot();
});

test('should handle remove button properly', () => {
   wrapper.find('button').at(0).simulate('click');
   expect(handleRemoveClick).toHaveBeenCalled();
   expect(closeModal).toHaveBeenCalled();
});

test('should handle no button click properly', () => {
   wrapper.find('button').at(1).simulate('click');
   expect(closeModal).toHaveBeenCalled();
});