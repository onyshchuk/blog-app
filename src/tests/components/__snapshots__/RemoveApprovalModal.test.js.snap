// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RemoveApprovalModal properly 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="Modal"
  closeTimeoutMS={200}
  contentLabel="Remove 'undefined' ?"
  isOpen={false}
  onRequestClose={[MockFunction]}
  overlayClassName="Overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3>
    Remove 
     ?
  </h3>
  <div
    className="buttons"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Remove
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      no
    </button>
  </div>
</Modal>
`;
